docs/src/main/asciidoc/se/injection.adoc

```diff
@@ -0,0 +1,850 @@
+link:{service-registry-base-url}/io/helidon/service/registry/Interception.ExternalDelegate.html[`@Interception.ExternalDelegate`]
+is designed for classes that you don’t have control over.
+This means it allows you to apply the interception mechanism even to third-party classes.
+
+It is not required to apply this annotation on the interfaces, however, it needs to be present for classes.
+
+Let's make the same `@Traced` annotation and Interceptor as in the previous examples
+
+[source,java]
+----
+include::{sourcedir}/se/inject/InterceptorDelegateExample.java[tag=snippet_1, indent=0]
+----
+
+In this example, we can’t change the class we’re intercepting in any way.
+Let's assume we have this external class, and we need to add an interception mechanism to it.
+
+[source,java]
+----
+include::{sourcedir}/se/inject/InterceptorDelegateExample.java[tag=snippet_4, indent=0]
+----
+
+Now we need to apply the
+link:{service-registry-base-url}/io/helidon/service/registry/Interception.ExternalDelegate.html[`@Interception.ExternalDelegate`]
+annotation on the factory class. Once this is done, interception will work as expected
+
+[source,java]
+.ExternalDelegate annotation used on the factory
+----
+include::{sourcedir}/se/inject/InterceptorDelegateExample.java[tag=snippet_5, indent=0]
+----
+
+== Events
+Events enable in-application communication between services by providing a mechanism to emit events and register consumers to handle them.
+
+A single event can be delivered to zero or more consumers.
+
+Key Terminology:
+
+- *Event Object* – Any object that is sent as an event.
+- *Event Emitter* – A service responsible for emitting events into the event system. Emitters are always generated.
+- *Event Producer* – A service that triggers an event by calling an emitter.
+- *Event Observer* – A service that listens for events, with a method annotated using
+link:{service-registry-base-url}/io/helidon/service/registry/Event.Observer.html[`@Event.Observer`].
+- *Qualified Event* – An event emitted with a qualifier, using an annotation marked with
+link:{service-registry-base-url}/io/helidon/service/registry/Service.Qualifier.html[`@Service.Qualifier`].
+
+=== Event Object
+To begin emitting events, the first step is to define the event type itself.
+
+[source,java]
+.Create a desired event type
+----
+include::{sourcedir}/se/inject/EventsExample.java[tag=snippet_1, indent=0]
+----
+
+=== Event Emitter
+Event emitters are code generated by Helidon. The emitter is generated when an injection point is discovered that expects it.
+
+All the emitters are generated as implementations of the
+link:{service-registry-base-url}/io/helidon/service/registry/Event.Emitter.html[`Event.Emitter`] interface.
+
+=== Event Producer
+An event producer is a service that triggers the event by using the event emitter.
+
+To emit an event, inject the desired
+link:{service-registry-base-url}/io/helidon/service/registry/Event.Emitter.html[`Event.Emitter`]
```

Comment:
The example and text contains too many `emit` words. Maybe rename the method in the event producer to `publish` or `produce` to get rid of at least one of them.